### PR TITLE
migrate kubelet configuration to edged kubelet configuration

### PR DESCRIPTION
**Pull request by Shelley-BaoYue**
**What type of PR is this?**

/kind feature
/kind api-change

**What this PR does / why we need it**:

In new feature  edged #4157 of v1.12 version, we import kubelet configuration directly. It is not convenient for us  to managing edged parameters, including tailoring, modification and defining. 

In next version, we will define a set of parameters `TailoredKubeletConfiguration` which KubeEdge need, this parameters are obtained by tailoring the Kubelet configuration and modified properly.

### CODE REVIEW

File: pkg/apis/componentconfig/edgecore/v1alpha2/types.go [hunk 1]
```diff
@@ -119,6 +120,547 @@ type Edged struct {
 	RegisterNodeNamespace string `json:"registerNodeNamespace,omitempty"`
 }
 
+// TailoredKubeletConfiguration indicates the tailored kubelet configuration
+type TailoredKubeletConfiguration struct {
```
Comment (wackxu) [inline pkg/apis/componentconfig/edgecore/v1alpha2/types.go:124]: please add comment for the config

```
TailoredKubeletConfiguration indicates the tailored kubelet configuration. It is derived from Kubernetes code `KubeletConfiguration` in package `k8s.io/kubelet/config/v1beta1` and made some variant. 
```

File: pkg/apis/componentconfig/edgecore/v1alpha2/default_kubelet_configuration.go [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+	componentbaseconfigv1alpha1 "k8s.io/component-base/config/v1alpha1"
+	kubeletconfigv1beta1 "k8s.io/kubelet/config/v1beta1"
+	"k8s.io/kubernetes/pkg/kubelet/qos"
+	kubetypes "k8s.io/kubernetes/pkg/kubelet/types"
+	utilpointer "k8s.io/utils/pointer"
+
+	"github.com/kubeedge/kubeedge/common/constants"
+)
+
+const (
```
Comment (wackxu) [inline pkg/apis/componentconfig/edgecore/v1alpha2/default_kubelet_configuration.go:16]: Why not just import these constants from k8s code?

File: pkg/apis/componentconfig/edgecore/v1alpha2/default_kubelet_configuration.go [hunk 3]
```diff
@@ -0,0 +1,99 @@
+package v1alpha2
```
[the remark this answers is not in the document]
Comment (Shelley-BaoYue) [reply pkg/apis/componentconfig/edgecore/v1alpha2/default_kubelet_configuration.go:24]: fixed.

File: pkg/apis/componentconfig/edgecore/v1alpha2/types.go (hunk 1 above)
Comment (Shelley-BaoYue) [reply pkg/apis/componentconfig/edgecore/v1alpha2/types.go:125]: > please update comment for the config
> 
> ```
> TailoredKubeletConfiguration indicates the tailored kubelet configuration. It is derived from Kubernetes code `KubeletConfiguration` in package `k8s.io/kubelet/config/v1beta1` and made some variant. 
> ```

updated.

File: pkg/apis/componentconfig/edgecore/v1alpha2/default_kubelet_configuration.go (hunk 2 above)
Comment (Shelley-BaoYue) [reply pkg/apis/componentconfig/edgecore/v1alpha2/default_kubelet_configuration.go]: > Why not just import these constants from k8s code?

fixed.

**Pull request merged**